### PR TITLE
Added kebechet administrator triggers to message investigators

### PR DESCRIPTION
**Pull request by saisankargochhayat**
## Related Issues and Dependencies
Related - https://github.com/thoth-station/workflow-helpers/issues/148
Closes https://github.com/thoth-station/investigator/issues/314

### CODE REVIEW

**Comment by saisankargochhayat** [comment]
@pacospace What do we want to be done with CVEProvidedMessage apart from scheduling Kebechet administrator? cc  - @bissenbay

**Comment by pacospace** [comment]
> @pacospace What do we want to be done with CVEProvidedMessage apart from scheduling Kebechet administrator? cc - @bissenbay

when a new CVE appear for a user software stack, I want to re run thamos advise and provide new stack! the message when we open the PR should clearly state, new CVE found in one dependency in your software stack, here is the new stack.

if the user has security recommendation, adviser might not find a stack with no CVEs so in that case it might fail, we should go back to the user and say there is a CVE for your dependency in your software stack, but we cannot find a stack with no CVEs, think of using similar packages (maybe new feature for package similarity might be good in the future).

@goern @fridex WDYT?

File: investigator/investigator/common.py [hunk 1]
```diff
@@ -76,11 +73,10 @@ async def wait_for_limit(openshift: OpenShift, workflow_namespace: str):
...
 
 
-async def schedule_kebechet_run_url(openshift: OpenShift, repo: str, gitservice_repo_name: str) -> int:
-    """Schedule Kebechet from repo url and service."""
-    # TODO: Add method to schedule kebechet
-
-    return 0
+async def schedule_kebechet_administrator(openshift: OpenShift, message_info: dict, message_name: str) -> int:
+    """Schedule Kebechet Administrator from a particular message."""
+    workflow_id = openshift.schedule_kebechet_administrator(message_info=message_info, message_type=message_name)
```
Comment (pacospace) [inline investigator/investigator/common.py:78]: it is missing `await wait_for_limit(openshift, workflow_namespace=Configuration.THOTH_BACKEND_NAMESPACE)`

File: investigator/investigator/missing_package/investigate_missing_package.py [hunk 2]
```diff
@@ -37,30 +36,11 @@
 @register_handler(MissingPackageMessage().topic_name, ["v1"])
 async def parse_missing_package(package: Dict[str, Any], openshift: OpenShift, graph: GraphDatabase):
     """Process a missing package message from package-update producer."""
-    repositories = graph.get_adviser_run_origins_all(
-        index_url=package["index_url"], package_name=package["package_name"], count=None, distinct=True,
-    )
+    # TODO call kebechet with the missing package name, which would call kebechet on the individual
```
Comment (pacospace) [inline investigator/investigator/missing_package/investigate_missing_package.py:39]: Here should be calling kebechet administrator as well right?

Comment (saisankargochhayat) [reply investigator/investigator/missing_package/investigate_missing_package.py:39]: We would need implement way to handle this on kebechet side, and then comeback to this.

File: investigator/investigator/common.py (hunk 1 above)
Comment (saisankargochhayat) [reply investigator/investigator/common.py]: Thanks, added check for limit

File: investigator/investigator/hash_mismatch/investigate_hash_mismatch.py [hunk 3]
```diff
@@ -64,23 +64,18 @@ async def parse_hash_mismatch(mismatch: Dict[str, Any], openshift: OpenShift, gr
...
-    issue_title = (
-        f"Hash mismatch for {mismatch['package_name']}=={mismatch['package_version']} on {mismatch['index_url']}"
-    )
-
-    def issue_body():
-        return "Automated message from package change detected by thoth.package-update"
+        # We schedule Kebechet Administrator workflow here -
+        workflow_id = await schedule_kebechet_administrator(
+            openshift=openshift, message_info=message_info, message_name=HashMismatchMessage.__name__,
+        )
```
Comment (pacospace) [inline investigator/investigator/hash_mismatch/investigate_hash_mismatch.py:77]: don't we miss the metric here?

File: investigator/investigator/missing_package/investigate_missing_package.py (hunk 2 above)
Comment (pacospace) [reply investigator/investigator/missing_package/investigate_missing_package.py:39]: Did you create issue in Kebechet? If not please add its :) Thanks @saisankargochhayat :)

File: investigator/investigator/hash_mismatch/investigate_hash_mismatch.py (hunk 3 above)
Comment (saisankargochhayat) [reply investigator/investigator/hash_mismatch/investigate_hash_mismatch.py:77]: Thanks I have added it.

File: investigator/investigator/missing_package/investigate_missing_package.py (hunk 2 above)
Comment (pacospace) [reply investigator/investigator/missing_package/investigate_missing_package.py:39]: I would deactivate the metric then for now :)

Comment (saisankargochhayat) [reply investigator/investigator/missing_package/investigate_missing_package.py:40]: @pacospace thanks I have commented it out for now.

File: investigator/investigator/cve_provided/investigate_cve_provided.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+from thoth.common import OpenShift
+from thoth.storages import GraphDatabase
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@count_exceptions(cve_provided_exceptions)
+@track_inprogress(cve_provided_in_progress)
+@register_handler(CVEProvidedMessage().topic_name, ["v1"])
+async def parse_cve_provided(cve_provided: Dict[str, Any], openshift: OpenShift, graph: GraphDatabase):
```
Comment (pacospace) [inline investigator/investigator/cve_provided/investigate_cve_provided.py:41]: do we do need `graph` here?

File: investigator/investigator/missing_package/investigate_missing_package.py [hunk 5]
```diff
@@ -37,30 +37,11 @@
...
-        requirements = re.split("\n| ", gitservice_repo.service.get_project().get_file_content("Pipfile"))
-
-        if package["package_name"] in requirements:
-            gitservice_repo.open_issue_if_not_exist(issue_title, issue_body)
-        else:
-            is_scheduled = await schedule_kebechet_run_url(
-                openshift=openshift, repo=repo, gitservice_repo_name=gitservice_repo.service_type.name
-            )
-            kebechet_wf_scheduled += is_scheduled
+    # TODO call kebechet with the missing package name, which would call kebechet on the individual
```
Comment (pacospace) [inline investigator/investigator/missing_package/investigate_missing_package.py:40]: do we do need `graph` in the method here?

**Review by pacospace — COMMENTED** [review]
Two minor comments and we are ready I think! Thank you

File: investigator/investigator/missing_package/investigate_missing_package.py (hunk 5 above)
Comment (saisankargochhayat) [reply investigator/investigator/missing_package/investigate_missing_package.py:38]: Removed these for now, I am not sure what we want to do apart from scheduling so had them there.

File: investigator/investigator/cve_provided/investigate_cve_provided.py (hunk 4 above)
Comment (saisankargochhayat) [reply investigator/investigator/cve_provided/investigate_cve_provided.py]: Currently, we don't not sure if we we wanna do anything apart from scheduling Kebechet.

File: investigator/investigator/missing_package/investigate_missing_package.py (hunk 5 above)
Comment (pacospace) [reply investigator/investigator/missing_package/investigate_missing_package.py:38]: In theory we want investigator really to do the minimum, aka schedling things, graph connections should be handled by components directly as much as possible

**Review by pacospace — APPROVED** [review]
Very nice work!! :100: :rocket:

**Pull request merged**